asset-transfer-sbe/README.md

```diff
@@ -0,0 +1,56 @@
+
+
+Fabric Shim Method's for State-Based endorsements (below references from fabric-shim, ChaincodeStub for Node):
+- setStateValidationParameter(key: string, ep: Uint8Array): Promise<void>;
+- getStateValidationParameter(key: string): Promise<Uint8Array>;
+- setPrivateDataValidationParameter(collection: string, key: string, ep: Uint8Array): Promise<void>;
+- getPrivateDataValidationParameter(collection: string, key: string): Promise<Uint8Array>;
+            
+
+## Testing the SBE Sample
+This sample can be tested with the fabric-samples, Test Network. To bootstrap the Test Network visit the [Test Network documentation](https://hyperledger-fabric.readthedocs.io/en/latest/test_network.html).
+
+Test network is a 2 Org network (Org1 and Org2). The endorsement policy for chaincodes deployed in this network defaults to majority, that is both Org's need to endorse transactions on the network.
+Using this sample, we demonstrate State-Based Endorsements (SBE), where it is possible to set a different endorsement policy for Asset Key. 
+
+On Asset creation, endorsement policy for the Asset is set to either Org1 or Org2, depending on the client creating the Asset. Creation of Asset still needs the default chaincode endorsement policy, and hence will need to be endorsed by both Org's (Org1 & Org2). However, all future updates for the Asset will need the modified endorsements as per the State-based endorsements set during Asset creation for the specified Key, instead of the default chaincode endorsement policy.
+
+Sample invokations to demonstrate State-Based Endorsements (SBE), on test-network
+* Create Asset (requires Org1 & Org2 Peer to endorse as per the default chaincode endorsement policy, however future updates will need only Org1 Peer to endorse, as the State-based endorsement policy is set for Key during the createAsset transaction)
+```bash
+peer chaincode invoke -o localhost:7050 --ordererTLSHostnameOverride orderer.example.com --tls --cafile ${PWD}/organizations/ordererOrganizations/example.com/orderers/orderer.example.com/msp/tlscacerts/tlsca.example.com-cert.pem -C mychannel -n asset-transfer-sbe --peerAddresses localhost:7051 --tlsRootCertFiles ${PWD}/organizations/peerOrganizations/org1.example.com/peers/peer0.org1.example.com/tls/ca.crt --peerAddresses localhost:9051 --tlsRootCertFiles ${PWD}/organizations/peerOrganizations/org2.example.com/peers/peer0.org2.example.com/tls/ca.crt -c '{"function":"CreateAsset","Args":["asset1","1","Foo"]}'
+```
+* Read & Verify Asset (Use this to verify changes for UpdateAsset & TransferAsset)
+```bash
+peer chaincode query -C mychannel -n asset-transfer-sbe -c '{"Args":["ReadAsset","asset1"]}'
+```
+* Update Asset (requires Org1 Peer to endorse as per SBE specified for assetId, but Org2 Peer is endorsing below, hence will result ENDORSEMENT_POLICY_FAILURE errors at commit, check peer logs)
```

Comment:
should be "hence will result in error ENDORSEMENT_POLICY_FAILURE during endorsement time"

Reply:
Since endorsement proposals will good from peer Orgs, and this will only be an issue when peer receives the transaction and invokes validation system chaincode associated for the transaction, hence modified as below:
"it will result in ENDORSEMENT_POLICY_FAILURE by the peer during endorsement validations for the transaction"